### ISSUE TITLE
JDA - website fancybox url links should check for IA source.

### ISSUE THREAD

**Issue by jamesburns** [Zeega/Zeega#463]

**Comment by kmlawson** [comment]
look in the seed database at the global table:

global->name=last_export_stat (gid=3)->value

This "value" gives you the seed ID of the most recently exported (and sent to Internet Archive) seed. Any seed ID>last_export_stat will NOT be in Internet Archive and should probably not be included in the search results.

**Comment by kanarinka** [comment]
What does this mean? Does it mean the "view on internet archive" link in fancybox should point to the URI with http://wayback-it/ in front? If so, maybe this is better to adjust in the data from the server as it's a distinction between uri vs attribution uri and we link to attribution uri for all the other fancybox media types.

**Comment by kmlawson** [comment]
"view on internet archive" should point to wayback link. If you point to the original "live link" then you aren't linking to the "Internet Archive" - if you like you can change the link from "view on internet archive" to "live link" if you like.

**Comment by jamesburns** [comment]
To clarify – the fancybox should check whether 'Website' type content has archive.org in the source url, if so it should say view on internet archive, otherwise it should say view live link.

**Comment by jamesburns** [comment]
From related issue:

In website fancybox wayback link at top links to earliest archived copy of the page at Internet Archive. However, the "View on Internet Archive" link below the preview of the archived website links to live version.

Recommended change in behavior:

show live link on top -below preview have three links:
Most Recent | Oldest | All

or the like...

To construct the links:
Most recent: http://wayback.archive-it.org/2438/9/[URL]
Oldest: http://wayback.archive-it.org/2438/20110301000000/[URL]
All: http://wayback.archive-it.org/2438/*/[URL]

To know whether a link has been submitted to Internet Archive for archiving or not (if it hasn't, the seed is likely not to be in the archive):

look in the seed database at the global table:

global->name=last_export_stat (gid=3)->value

This "value" gives you the seed ID of the most recently exported (and sent to Internet Archive) seed. Any seed ID>last_export_stat will NOT be in Internet Archive and should probably not be included in the search results.

**Comment by kanarinka** [comment]
If people have added a website from Internet Archive to the JDA archive, haven't they added a specific date (which we store in attribution_uri)? 

I'm unclear about how I can access the timestamp of the website in IA to construct urls for Oldest/Most Recent and also to know whether it's oldest or most recent or somewhere in between. I would suggest we have "View on Internet Archive" and "View all archived versions" (which would link to the list which I can get by parsing the attribution_uri).

**Comment by kmlawson** [comment]
Hmm,

We don't have control over the exact dates that IA harvests. 

It might help to understand how we submit the websites:

User submits seed URL to jdarchive.org/contribute together with website metadata

Every 3-4 weeks, I export all the new seeds into an Excel spreadsheet (I know...but that is the way they want it) and email it to Internet Archive. They then begin the harvest of these websites during the next few days (but we won't know when exactly). It then gets harvested at a frequency partly determined by us and partly, it seems, their server.

Unless you ping the IA server at http://wayback.archive-it.org/2438/20110301000000/[THE SEED URL] you can't know with 100% certainty if Internet Archive has harvested a copy of the website (they will have an error page if they haven't). 

However, there is a semi-reliable way to tell if a URL has been harvested or not: if you check the "last_export_stat" row in the "global" table of our seeds database you can see the most recently exported seed ID. Usually any exported seed will have been archived, or will very soon be archived by the Internet Archive.

In other words: timestamps are not useful to tell anything for these seeds. The Oldest / Most Recent / All can always be reached with:

Most recent: http://wayback.archive-it.org/2438/9/[URL]
Oldest: http://wayback.archive-it.org/2438/20110301000000/[URL]
All: http://wayback.archive-it.org/2438/*/[URL]

This is because the Oldest is March 1, 2011. IA will return the first harvest after March 1 (before we started harvesting for our collection) and "9" is an indication to IA that you want the most recent harvest, while "*" is an indication that you want the index. 

I hope this makes it clearer!

Here is some of the quick and dirty code we used to create these links on our seed database at http://jdarchive.org/seeds/ :

```
function last_exported_seed() {
    $array = array();
    $array['timestamp'] = '';
    $array['sid'] = '';
    $sql = "SELECT value, timestamp FROM global WHERE name = 'last_export_start'";
    $result = mysql_query($sql);

    if (mysql_num_rows($result) == 1) {
      $row = mysql_fetch_row($result);
      $array['timestamp'] = $row[1];
      $array['sid'] = $row[0];
    }

    return $array;
  }

```

....

```
    $scope = $seed->scope;
    $frequency = $seed->frequency;
    $iaurl=$seed->url;
    echo " <a href='$iaurl'>Live Link</a> &nbsp;&nbsp;";
    if ($last_exported['sid']<$seed->sid) {
        echo "<strong>Not Yet Submitted</strong>&nbsp;&nbsp;";
    } else {
    echo "<strong>Archived Copies: <a href='http://wayback.archive-it.org/2438/9/$iaurl'>Most Recent</a> </strong> | <strong>
    <a href='http://wayback.archive-it.org/2438/20110301000000/$iaurl'>Earliest</a> </strong>
    | <strong>
    <a href='http://wayback.archive-it.org/2438/*/$iaurl'>All</a> &nbsp;&nbsp;";
    }
    echo " Scope: </strong>".ucfirst($scope)." &nbsp;&nbsp;<strong>Frequency:</strong> ".ucfirst($frequency)."
    "; 
```

```
```

**Comment by jamesburns** [comment]
On hold for now, while we clean up other bugs.  We will spec out exactly what this should look like and then implement.
